src/main/java/teammates/ui/controller/InstructorCourseInstructorDeleteAction.java

```diff
@@ -27,6 +28,15 @@ protected ActionResult execute() {
 
         /* Process deleting an instructor and setup status to be shown to user and admin */
         if (hasAlternativeInstructor(courseId, instructorEmail)) {
+            List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+            isError = !logic.isAtLeastOneInstructorVisibleToStudents(instructors, instructorEmail);
```

Comment:
It seems like you're query-ing a list of instructors for the course only to find if there is a match with the supplied `instructorEmail`. A large majority of this query is wasted. Is it possible to create a method in `InstructorsDb` (and subsequent methods in `InstructorsLogic`, `Logic`) that does as minimum database query as possible?